transport/processing/processing.py

```diff
@@ -0,0 +1,332 @@
+
+    uploaders = [instantiate_uploader(upload_cfg) for upload_cfg in upload_cfgs]
+
+    pipeline = ProcessingPipeline(
+        name=name,
+        filters=filters,
+        downloader=downloader,
+        processor=proc,
+        uploaders=uploaders,
+    )
+
+    return pipeline
+
+
+def enum_processing_cfgs(
+    processing_cfg: dict,
+    shared_processors: dict,
+    shared_uploaders: dict,
+):
+    cfg_entries = processing_cfg['processing_cfg']
+
+    yield from map(
+        processing_pipeline,
+        cfg_entries,
+        itertools.repeat(shared_processors, len(cfg_entries)),
+        itertools.repeat(shared_uploaders, len(cfg_entries)),
+    )
+
+
+def create_jobs(processing_cfg, component_descriptor):
+    shared_processors = {
+        name: _processor(cfg) for name, cfg in processing_cfg.get('processors', {}).items()
+    }
+    shared_uploaders = {
+        name: _uploader(cfg) for name, cfg in processing_cfg.get('uploaders', {}).items()
+    }
+
+    for component, container_image in _enumerate_oci_resources(component_descriptor):
+        for processor in enum_processing_cfgs(
+            processing_cfg,
+            shared_processors,
+            shared_uploaders,
+        ):
+
+            job = processor.process(component=component, container_image=container_image)
+            if not job:
+                continue # processor did not want to process
+
+            ci.util.info(
+                f'found matching processor: {component.name}:{container_image.access.imageReference}'
+            )
+            yield job
+            break
+        else:
+            ci.util.warning(
+                f'no matching processor: {component.name}:{container_image.access.imageReference}'
+            )
+
+
+def _enumerate_oci_resources(descriptor):
+    for resource in itertools.chain(
+            descriptor.component.externalResources,
+            descriptor.component.localResources,
+        ):
+        if resource.access.type == cm.AccessType.OCI_REGISTRY and \
+           resource.type == cm.ResourceType.OCI_IMAGE:
+            yield (descriptor.component, resource)
+
+
+def process_resources(processing_cfg, component_obj):
+    if config.DRY_RUN:
+        ci.util.warning('dry-run: not downloading or uploading any images')
+
+    src_descriptor = component_obj.descriptor
+    src_component = src_descriptor.component
+
+    executor = concurrent.futures.ThreadPoolExecutor(max_workers=8)
+
+    jobs = create_jobs(
+        processing_cfg=processing_cfg,
+        component_descriptor=src_descriptor,
+    )
+
+    NewResources = collections.namedtuple(
+        'NewResources',
+        ['resources', 'expected']
+    )
+
+    # only modify the OCI_REGISTRY resources
+    def _new_resources(resources: cm.AccessType) -> NewResources:
+        return NewResources(
+            resources=list(r for r in resources if r.access.type != cm.AccessType.OCI_REGISTRY),
+            expected=len(resources)
```

Comment:
Maybe naming it something like `expected_resource_count` or something like that would be more explicit - not sure.
@ccwienk is usually better at naming stuff than me.

Also, could you point me to where that is incremented?

Reply:
additional nit: rather use a `list-comprehension` (square brackets []), rather than a generator-comprehension and immediate conversion into a list.